src/lib/services/state-service.ts

```diff
@@ -200,6 +200,30 @@ export default class StateService {
                 dropBeforeImport,
                 keepExisting,
             });
+
+            if (
+                featureEnvironments
+                    .map((fe) => fe.environment)
+                    .filter((eName) => eName !== 'default').length > 0
+            ) {
```

Comment:
This is hairy: we want to create this link only in very specific cases. We can't rely on the version number, but this situation is likely to happen if you have multiple environments. I'd like some input on this one @chriswk and @sighphyre

Reply:
So, in the end, there was an issue with my logic to find whether the feature should be enabled in the environment or not, which I found thanks to @chriswk suggestion to refactor the code. After I fixed that, I can now remove the restriction I had here